rcl/test/rcl/test_graph.cpp

```diff
@@ -852,6 +852,67 @@ check_graph_state(
   }
 }
 
+void
+check_entity_number(
+  const rcl_node_t * node_ptr,
+  std::string & topic_name,
+  size_t expected_publisher_count,
+  size_t expected_subscriber_count,
+  bool expected_in_tnat,
+  std::chrono::seconds timeout)
+{
+  RCUTILS_LOG_DEBUG_NAMED(
+    ROS_PACKAGE_NAME,
+    "Expecting number of %zu publishers, %zu subscribers, and that the topic is%s in the graph.",
+    expected_publisher_count,
+    expected_subscriber_count,
+    expected_in_tnat ? "" : " not"
+  );
+  bool is_in_tnat = false;
+  rcl_names_and_types_t tnat {};
+  rcl_ret_t ret;
+  rcl_allocator_t allocator = rcl_get_default_allocator();
+  size_t pub_count, sub_count;
+
+  // Check number of entities until timeout expires.
+  auto start_time = std::chrono::system_clock::now();
+  do {
+    ret = rcl_count_publishers(node_ptr, topic_name.c_str(), &pub_count);
+    ASSERT_EQ(ret, RCL_RET_OK);
+    ret = rcl_count_subscribers(node_ptr, topic_name.c_str(), &sub_count);
+    ASSERT_EQ(ret, RCL_RET_OK);
+    if ((expected_publisher_count == pub_count) &&
+      (expected_subscriber_count == sub_count))
+    {
+      break;
+    }
+    std::this_thread::sleep_for(std::chrono::milliseconds(100));
+  } while (std::chrono::system_clock::now() - start_time < timeout);
+  EXPECT_EQ(expected_publisher_count, pub_count);
+  EXPECT_EQ(expected_subscriber_count, sub_count);
+
+  tnat = rcl_get_zero_initialized_names_and_types();
+  ret = rcl_get_topic_names_and_types(node_ptr, &allocator, false, &tnat);
+  ASSERT_EQ(RCL_RET_OK, ret) << rcl_get_error_string().str;
+  is_in_tnat = false;
+  for (size_t i = 0; i < tnat.names.size; ++i) {
```

Comment:
We can leave it like how it was in `check_graph_state` for early termination? Or was there a specific reason you removed the `ret` check?
```suggestion
  for (size_t i = 0; RCL_RET_OK == ret && i < tnat.names.size; ++i) {
```

Reply:
`ret` is checked a line before with `ASSERT_EQ`, this guarantees `ret` is `RCL_RET_OK`, otherwise test assertion fails and will never get to be this `for` loop.